postman-generator/src/main/scala/generator/DependantOperationResolver.scala

```diff
@@ -1,83 +1,123 @@
+
+    for {
+      attribute <- allAttributes
+      operation <- findOperationForAttribute(attribute, serviceNamespaceToResources).toSeq
+    } yield {
+      (attribute, operation)
     }
-      .collect {
-        case (objRefAttr, Some(operation)) =>
-          (objRefAttr, operation)
+  }
+
+  private def findReferenceAttributeInModelField(field: Field): Option[ModelReference] = {
+
+    field.attributes.collectFirst {
+      case attr if attr.name.equalsIgnoreCase(AttributeName.ObjectReference.toString) =>
+        attr.value.asOpt[ModelReference]
+    }.flatten
+  }
+
+  private def findOperationForAttribute(objRefAttr: ExtendedObjectReference, serviceNamespaceToResources: Map[String, Seq[Resource]]) = {
+    serviceNamespaceToResources
+      .getOrElse(key = objRefAttr.relatedServiceNamespace, default = Nil)
+      .filter(_.`type` === objRefAttr.resourceType)
+      .flatMap(_.operations)
+      .find(_.method === objRefAttr.operationMethod)
+  }
+
+  private def deepSearchModelsForAttributes(typ: String, service: Service): Seq[ModelReference] = {
+
+      def recurSearch(typ: String): Seq[ModelReference] = {
+        service.models.find(_.name === typ) match {
+          case Some(model) =>
+            model.fields.flatMap {
+              // model have fields with another models, going deeper
+              case field if service.models.exists(_.name === field.`type`) =>
+                recurSearch(field.`type`)
+              // model is a "leaf", searching for special attribute
+              case field =>
+                val objRefAttrOpt = findReferenceAttributeInModelField(field)
+
+                objRefAttrOpt match {
+                  case Some(objAttrRef) =>
+                    val modelToLookup =
+                      if (objAttrRef.relatedServiceNamespace != service.namespace)
+                        s"${objAttrRef.relatedServiceNamespace}.models.${objAttrRef.resourceType}"
+                      else
+                        objAttrRef.resourceType // namespace from the main service, plain model name
+                    // adding and going recursive
+                    recurSearch(modelToLookup) :+ objAttrRef
+                  case None =>
+                    Nil
+                }
+            }
+          case None if service.unions.exists(_.name === typ) =>
+            val union = service.unions.find(_.name === typ).get
+            union.types.flatMap(u => recurSearch(u.`type`))
+          case _ =>
+            Nil
+        }
       }
 
-    objRefAttrToOperation.distinct
+      recurSearch(typ)
+  }
+
+  private def findParamNameInPathString(path: String): Option[String] = {
+    val regex = """\:(\w+)[\/]{0,1}""".r
+    regex
+      .findAllIn(path)
+      .map(str => regex.replaceAllIn(str, "$1"))
+      .toList
+      .headOption //TODO investigate few params like ":organization/order/:id" in one resource path. Is it valid/used anywhere ?
```

Comment:
regarding this `TODO`. I don't know if I understand it correctly.
There are 2 ids in one URL. it's quite common, when you query for a single item.
for example
https://github.com/flowcommerce/api/blob/master/spec/catalog.json#L1464

does it affect the solution anyhow?